### PR TITLE
Adding time series support to make_pipeline

### LINKED ISSUE #1497: Update make_pipelines to work for timeseries pipelines

### PR DESCRIPTION
**Pull request by freddyaboulton**
### Pull Request Description
Fixes #1497

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

File: evalml/pipelines/utils.py [hunk 1]
```diff
@@ -66,9 +67,11 @@ def _get_preprocessing_components(X, y, problem_type, text_columns, estimator_cl
     if add_datetime_featurizer:
         pp_components.append(DateTimeFeaturizer)
 
-    # DateTimeFeaturizer can create categorical columns
+    if problem_type in [ProblemTypes.TIME_SERIES_REGRESSION]:
+        pp_components.append(DelayedFeatureTransformer)
+
     categorical_cols = X.select('category')
-    if (add_datetime_featurizer or len(categorical_cols.columns) > 0) and estimator_class not in {CatBoostClassifier, CatBoostRegressor}:
+    if len(categorical_cols.columns) > 0 and estimator_class not in {CatBoostClassifier, CatBoostRegressor}:
```
Comment (freddyaboulton) [inline evalml/pipelines/utils.py:74]: `DateTimeFeaturizer` no longer creates categorical features by default.

File: evalml/tests/pipeline_tests/test_pipelines.py [hunk 2]
```diff
@@ -111,27 +113,30 @@ def test_make_pipeline_all_nan_no_categoricals(input_type, problem_type):
...
 
     estimators = get_estimators(problem_type=problem_type)
-    if problem_type == ProblemTypes.BINARY:
-        pipeline_class = BinaryClassificationPipeline
-    elif problem_type == ProblemTypes.MULTICLASS:
-        y = pd.Series([0, 2, 1, 2, 0])
-        pipeline_class = MulticlassClassificationPipeline
-    elif problem_type == ProblemTypes.REGRESSION:
-        pipeline_class = RegressionPipeline
+    pipeline_class = _get_pipeline_base_class(problem_type)
```
Comment (bchen1116) [inline evalml/tests/pipeline_tests/test_pipelines.py:116]: nice!

**Review by bchen1116 — APPROVED** [review]
LGTM! Nice tests!

File: evalml/pipelines/utils.py [hunk 3]
```diff
@@ -66,9 +67,11 @@ def _get_preprocessing_components(X, y, problem_type, text_columns, estimator_cl
     if add_datetime_featurizer:
         pp_components.append(DateTimeFeaturizer)
 
-    # DateTimeFeaturizer can create categorical columns
+    if problem_type in [ProblemTypes.TIME_SERIES_REGRESSION]:
```
[the remark this answers is not in the document]
Comment (freddyaboulton) [reply evalml/pipelines/utils.py:70]: I'm about to add time series classification to the list! I don't know if the better thing is a `is_time_series` method to the problem types enum?

[the remark this answers is not in the document]
Comment (bchen1116) [reply evalml/pipelines/utils.py:70]: Got it. This makes sense then. Thanks!

**Pull request merged**